src/XMakeBuildEngine/Definition/Project.cs

```diff
@@ -1167,6 +1176,7 @@ public List<ProvenanceResult> GetItemProvenance(string itemToMatch)
         /// <param name="itemType">The item type to constrain the search in</param>
         public List<ProvenanceResult> GetItemProvenance(string itemToMatch, string itemType)
         {
+            ReevaluateForPostMortemAnalysisIfNecessary();
```

Comment:
I don't love having to do this manually in all the right places, but I can't think of a better way at the moment.